src/annotator/sidebar.js

```diff
@@ -315,6 +293,27 @@ export class Sidebar {
     sidebarTrigger(document.body, () => this.open());
     features.init(this._sidebarRPC);
 
+    this._sidebarRPC.on('connect', () => {
+      // Show the UI
+      if (this.iframeContainer) {
+        this.iframeContainer.style.display = '';
+      }
+
+      // Set initial highlight visibility. We do this only once the sidebar app
+      // is ready because `setHighlightsVisible` needs to reflect this state to
+      // the sidebar app.
```

Comment:
Given that we can now queue messages to be sent to the sidebar app before it has actually loaded, I wonder if this comment is still valid.

Reply:
I agree it is not relevant anymore. I personally think that the state for toggling the highlights visibility should be the host frame. Currently, there is an indirection that I don't think it is necessary: host -> sidebar -> guests. It could be simplified as host -> guests.